azavea/cac-tripplanner
# Provisioning Fixes

## Overview

Updates to provisioning to get this project to setup. With `virtualbox` now being the default for this app, we would no longer need to specify `CAC_APP_SHARED_FOLDER_TYPE=virtualbox` before running `vagrant` commands on Windows and Linux.

### Notes

By tying our NPM version to Ubuntu we make it less explicit, and potentially risk having it change when we do an Ubuntu upgrade. However, it was the easiest way to get a reliably working environment, and since the associated Ansible Role is old and unmaintained, I think it's alright.

## Testing Instructions

- Check out this branch
- Remove your local `node_modules` folder
    ```bash
    rm -rf src/node_modules
    ```
- Delete and reprovision the `app` VM
    ```bash
    vagrant destroy -f app && vagrant up app
    ```
  - [x] Ensure it provisions correctly
- Build the front-end
    ```bash
    ./scripts/serve-js-dev.sh
    ```
- Go to http://localhost:8024/
  - [x] Ensure you see the running app

## Checklist

Connects #XXX
